### PR TITLE
HAWQ-1506. Support multi-append a file within encryption zone

### PR DESCRIPTION
**Pull request by interma**
This bug is happened on: 
1, append a existed file. 
2, call multi `hdfsWrite()`

Test case please see `TEST(TestCInterfaceTDE, TestAppendMultiTimes)`.

### CODE REVIEW

**Comment by interma** [comment]
@linwen @wengyanqing @amyrazz44 @wcl14 help to review, thanks!

File: depends/libhdfs3/src/client/CryptoCodec.cpp [hunk 1]
```diff
@@ -25,154 +25,188 @@
...
+		if (is_init)
+			return 0;
+
+		// Get decrypted key from KMS
+		std::string key;
+		try {
+			key = getDecryptedKeyFromKms();
+		}
+		catch (...) {
+			LOG(WARNING, "getDecryptedKeyFromKms failed");
```
Comment (wengyanqing) [inline depends/libhdfs3/src/client/CryptoCodec.cpp]: It's better to throw the exception to upper function then return error.

Comment (interma) [inline depends/libhdfs3/src/client/CryptoCodec.cpp]: You are right, the exception should throw to upper caller to judge how to deal with it.
Thanks!

**Comment by wengyanqing** [comment]
LGTM.

**Comment by radarwave** [comment]
+1

**Comment by interma** [comment]
merged

**Pull request closed without merging**